File: examples/injectdemo/java/injectdemo/Message.java
// SPDX-License-Identifier: GPL-3.0-only
package injectdemo;

import javax.inject.Inject;

/**
 * All three injection kinds on one plain class: constructor, field, and method. Built from ordinary
 * code through the generated {@code Message_Factory.get()} — the entry point for pulling a graph
 * object outside a framework-owned component.
 */
public class Message {
  private final Clock ctorClock;

  @Inject Clock clock;

  private Greeter greeter;

  @Inject
  public Message(Clock clock) {
    this.ctorClock = clock;
  }

  @Inject
  void setGreeter(Greeter greeter) {
    this.greeter = greeter;
  }

  public boolean fieldsOk() {
    return clock != null && clock == ctorClock;
  }

  public boolean methodOk() {
    return greeter != null && greeter.clock() == clock;
  }
}

File: examples/injectdemo_kt/kotlin/injectdemokt/Message.kt
// SPDX-License-Identifier: GPL-3.0-only
package injectdemokt

import javax.inject.Inject

/**
 * All three injection kinds on one plain class: constructor, field, and method. Built from ordinary
 * code through the generated `Message_Factory.get()` — the entry point for pulling a graph object
 * outside a framework-owned component.
 */
class Message @Inject constructor(private val ctorClock: Clock) {
    @Inject lateinit var clock: Clock

    private var viaMethod: Greeter? = null

    @Inject
    fun setGreeter(greeter: Greeter) {
        viaMethod = greeter
    }

    fun fieldsOk(): Boolean = clock === ctorClock

    fun methodOk(): Boolean = viaMethod?.clock === clock
}
